presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -308,6 +318,30 @@ default TableStatisticsMetadata getStatisticsCollectionMetadata(ConnectorSession
         return TableStatisticsMetadata.empty();
     }
 
+    /**
+     * Describes statistics that must be collected during an ANALYZE statement.
+     */
+    default TableStatisticsMetadata getStatisticsCollectionMetadataForAnalyze(ConnectorSession session, ConnectorTableMetadata tableMetadata)
```

Comment:
This still couples the connector abstractions too closely to how the behavior is exposed at the language level. At the core, this is about computing statistics for a table. We should rename these methods to something that reflects that more closely. E.g., getTableStatisticsMetatdata, beginStatisticsCollection, finishStatisticsCollection. Also, remove references to "ANALYZE query" in the comments.